### PR TITLE
fhir-persistence-schema GRANT must consider resource #1169

### PR DESCRIPTION
**Pull request by prb112**
Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java [hunk 1]
```diff
@@ -497,7 +495,9 @@ protected void grantPrivileges(String groupName) {
         final IDatabaseAdapter adapter = getDbAdapter(dbType, connectionPool);
         try (ITransaction tx = TransactionFactory.openTransaction(connectionPool)) {
             try {
-                pdm.applyGrants(adapter, groupName, grantTo);
+                pdm.applyGrants(adapter, FhirSchemaConstants.FHIR_USER_GRANT_GROUP, grantTo);
+                pdm.applyGrants(adapter, FhirSchemaConstants.FHIR_OAUTH_GRANT_GROUP, grantTo);
+                pdm.applyGrants(adapter, FhirSchemaConstants.FHIR_BATCH_GRANT_GROUP, grantTo);
```
Comment (albertwang-ibm) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:500]: so, all the db objects will be processed 3 times - once for each group?  even thought maybe only some of them belong to certain group ...

Comment (lmsurpre) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:500]: well, if I understand it right, `buildCommonModel` will only add the "target" objects to the pdm, so its not necessarilly "all" the db objects.

Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:500]: it's not necessarily all, but yes, it'll potentially process three times, applying only where the group matches

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/FhirSchemaGenerator.java [hunk 2]
```diff
@@ -83,13 +83,6 @@
     private static final String ADD_RESOURCE_TYPE = "ADD_RESOURCE_TYPE";
     private static final String ADD_ANY_RESOURCE = "ADD_ANY_RESOURCE";
 
-    // Tags used to control how we manage privilege grants
```
Comment (prb112) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/FhirSchemaGenerator.java:86]: @punktilious below

**Comment by lmsurpre** [comment]
Robin helped me confirm that the GRANT_GROUP is never actually persisted, so I think that resolves my one concern

**Pull request merged**